File: python/typical90/typical90_ax.py
n,l=map(int,input().split())
MOD=10**9+7
dp=[0]*(n+1)
dp[0]=1
for i in range(n+1):
    if i+l<=n:
        dp[i+l]+=dp[i]
        dp[i+l]%=MOD
    if i+1<=n:
        dp[i+1]+=dp[i]
        dp[i+1]%=MOD
print(dp[n])

File: rust/typical90/src/bin/typical90_ax.rs
fn main() {
    proconio::input!{
        n:usize,
        l:usize
    }
    const MOD:usize = 1_000_000_007;
    let mut dp = vec![0;n+1];
    dp[0] = 1;
    for i in 0..=n {
        if i+l<=n {
            dp[i+l] += dp[i];
            dp[i+l] %= MOD;
        }
        if i+1<=n {
            dp[i+1] += dp[i];
            dp[i+1] %= MOD;
        }
    }
    println!("{}",dp[n]);
}
